### PR TITLE
chore(compass-query-history): Update to shared config COMPASS-5936

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-5936

Updates `compass-query-history` to the shared config. Fixes two lint issues (commented in pr w/ screenshots) and unskips a test we were skipping.

This work preps for updating the query history plugin's toolbar to leafygreen components and typescript.

I'll open a follow up pr with the `reformat` prettier changes to keep things more granular.

### CODE REVIEW

File: packages/compass-query-history/src/components/header/header.jsx [hunk 1]
```diff
@@ -50,20 +55,22 @@ class Header extends PureComponent {
...
+          className={closeButtonStyles}
           data-test-id="query-history-button-close-panel"
-          href="#"
-          onClick={this.collapse}>
-          ×
-        </span>
+          onClick={this.collapse}
+          aria-label="Close query history"
+        >
+          <Icon glyph="X" />
```
Comment (Anemy) [inline packages/compass-query-history/src/components/header/header.jsx:72]: Update to fix the non-interactive html tag lint issue.
Before:
Screen Shot 2022-07-06 at 3 20 52 PM

After:
Screen Shot 2022-07-06 at 3 22 07 PM

File: packages/compass-query-history/src/components/query/query.jsx [hunk 2]
```diff
@@ -34,32 +55,39 @@ class Query extends PureComponent {
...
+      <button
         onClick={this.populateQuery}
-        className={styles.component}
-        data-test-id="query-history-query-attributes">
+        className={queryAttributesContainerStyles}
+        data-test-id="query-history-query-attributes"
+      >
         { Object.keys(attributes).map(this.renderAttr) }
-      </ul>
+      </button>
```
Comment (Anemy) [inline packages/compass-query-history/src/components/query/query.jsx:90]: Updated to fix the non-interactive html tag lint issue.
Before:
Screen Shot 2022-07-06 at 3 45 01 PM

After:
Screen Shot 2022-07-06 at 3 44 54 PM

File: packages/compass-query-history/src/components/query/query.jsx [hunk 3]
```diff
@@ -34,32 +55,40 @@ class Query extends PureComponent {
...
+        data-test-id="query-history-query-attribute"
+        className={queryAttributeStyles}
+        key={index}
+      >
+        <Label
           data-test-id="query-history-query-label"
-          className={styles.label}>{attrKey}</label>
+          className={labelStyles}
+          htmlFor={`query-history-query-attr-${attrKey}`}
+        >{attrKey}</Label>
```
Comment (gribnoysup) [inline packages/compass-query-history/src/components/query/query.jsx:67]: This Code component you are referring in the for here is not an input, right? [HTML label elements have a very particular use-case and meaning](https://html.spec.whatwg.org/multipage/forms.html#the-label-element) and should provide labels for a specific set of elements

Comment (Anemy) [reply packages/compass-query-history/src/components/query/query.jsx]: Thanks for catching! Updated

**Review by gribnoysup — APPROVED** [review]
A few small things caught my eye, otherwise looks good

File: packages/compass-query-history/src/components/query/query.spec.jsx [hunk 4]
```diff
@@ -1,34 +1,35 @@
...
       const node = component.find('[data-test-id="query-history-query-attributes"]');
-      expect(node).to.have.type('ul');
+      expect(node).to.have.type('button');
     });
 
-    it('renders the attribute label', () => {
+    it('renders the attribute label', function() {
       const node = component.find('[data-test-id="query-history-query-label"]');
-      expect(node).to.have.text('filter');
+      expect(node).to.contain.html('>filter<');
```
[the remark this answers is not in the document]
Comment (Anemy) [reply packages/compass-query-history/src/components/query/query.spec.jsx]: Good call, updated to check the text directly (was shallow mounting before now we're mounting fully with enzyme).

**Pull request merged**